packages/remirror__extension-node-formatting/src/node-formatting-extension.ts

```diff
@@ -207,12 +208,20 @@ export class NodeFormattingExtension extends PlainExtension<NodeFormattingOption
     return this.rightAlign()(props);
   }
 
-  @keyBinding({ shortcut: NamedShortcut.IncreaseIndent, command: 'increaseIndent' })
+  @keyBinding({
+    shortcut: NamedShortcut.IncreaseIndent,
+    command: 'increaseIndent',
+    priority: ExtensionPriority.Low,
+  })
   increaseIndentShortcut(props: KeyBindingProps): boolean {
     return this.increaseIndent()(props);
   }
 
-  @keyBinding({ shortcut: NamedShortcut.DecreaseIndent, command: 'decreaseIndent' })
+  @keyBinding({
+    shortcut: NamedShortcut.DecreaseIndent,
+    command: 'decreaseIndent',
+    priority: ExtensionPriority.High,
```

Comment:
Why is this one High is the other is Low priority?

Reply:
- When pressing tab
  - Try to indent list item first (ExtensionPriority.Default)
  - If not possible, then indent node formatting (ExtensionPriority.Low)
- When pressing shift-tab
  - Try to dedent node formatting first (ExtensionPriority.High)
  - If not possible, then dedent list item (ExtensionPriority.Default)

Reply:
My idea is to make `tab` and `shift-tab` symmetric. 

For a pair of shortcuts which do opposite operations, by pressing one shortcut, you kind of add a new state into a stack, and pressing the other shortcut will pop the lastest state for the stack and make you have the previous state. 

We have many pairs of shortcuts that match this rule, for example Undo - Redo, Zoom In - Zoom Out, Left Arrow - Right Arrow. 

```
       --- KeyA -->        --- KeyA -->        --- KeyA --> 
state1              state2              state3              state4
       <-- KeyB ---        <-- KeyB ---        <-- KeyB ---
```

https://user-images.githubusercontent.com/24715727/167158142-f3bcd3d9-6aaf-4674-aca5-7c6e54b6f90e.mp4

Reply:
That wouldn't be true though right?

If I was within a list, in an indented parent 

```
blockquote(indent=20)
  bulletList
    listitem
      paragraph
    listItem
      paragraph<cursor>
```

and press tab, I indent list item. 

If I then press shift-tab I end up dedenting the entire parent blockquote? Seems wrong to me?

Reply:
Unless all the commands check only the current depth?

Reply:
@ocavue I think both our concerns could be addressed by having

When pressing shift-tab
* Try to dedent node formatting at **current depth only** (ExtensionPriority.Medium)
* Try to dedent list item (ExtensionPriority.Default)
* If not possible, then dedent node formatting of any parent (ExtensionPriority.Low)

Thoughts?